independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ListExtensionsCommandHandler.java

```diff
@@ -49,15 +51,19 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
         } catch (IOException e) {
             throw new QuarkusCommandException("Failed to determine the list of installed extensions", e);
         }
-
-        Stream<Extension> platformExtensionsStream = invocation.getPlatformDescriptor().getExtensions().stream();
-        platformExtensionsStream = platformExtensionsStream.filter(this::filterUnlisted);
-        if (search != null && !"*".equalsIgnoreCase(search)) {
-            final Pattern searchPattern = Pattern.compile(".*" + search + ".*", Pattern.CASE_INSENSITIVE);
-            platformExtensionsStream = platformExtensionsStream.filter(e -> filterBySearch(searchPattern, e));
+        final Collection<Extension> platformExtensions;
+        if (extensionRegistry != null) {
+            String quarkusVersion = invocation.getPlatformDescriptor().getQuarkusVersion();
+            platformExtensions = extensionRegistry.list(quarkusVersion, search);
+        } else {
+            Stream<Extension> platformExtensionsStream = invocation.getPlatformDescriptor().getExtensions().stream();
+            platformExtensionsStream = platformExtensionsStream.filter(this::filterUnlisted);
+            if (search != null && !"*".equalsIgnoreCase(search)) {
+                final Pattern searchPattern = Pattern.compile(".*" + search + ".*", Pattern.CASE_INSENSITIVE);
+                platformExtensionsStream = platformExtensionsStream.filter(e -> filterBySearch(searchPattern, e));
+            }
+            platformExtensions = platformExtensionsStream.collect(Collectors.toList());
```

Comment:
should list/add not also include the existing platform info in its search ?

Reply:
Existing platform info? Can you elaborate?

Reply:
the project on disk has a platform, shouldn't that platforms json be included in the search ? i.e. in cases where registry is broken, faulty or simply just not available in that registry.

Reply:
Ah good point, yeah, I'll do that